cdap-api/src/main/java/co/cask/cdap/api/service/http/NonTransactionalHttpContentConsumer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.service.http;
+
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.TxRunnable;
+import co.cask.cdap.api.dataset.Dataset;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A {@link HttpContentConsumer} that supports non-transactional processing of HTTP request body.
+ * One can extends from this class and implements the {@link #onReceived(ByteBuffer, Transactional)} method
```

Comment:
The idea is to have the method receives a Transactional instance, so that it can still optionally execute function inside a transaction if needed, but the call to the onReceive method is not inside a transaction.